### PR TITLE
Test to verify write operation should fail from secondary site in rbd mirroring

### PR DESCRIPTION
**Pull request by sunilangadi2**
# Description
Automated below test case
https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-9503

As discussed with team since it's just one negative test to validate, hence accommodate that test in existing 
`tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py` as it covers both validation from `journal` and `snapshot` based mirroring.

**Test flow:**
- On the remote/secondary cluster, try using the mirrored image for writing.
- command should error out saying `Read only file system`

**Files modified:**
tests/rbd_mirror/rbd_mirror_utils.py
tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py

**Test Result:**
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-DMXXLM

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py [hunk 1]
```diff
@@ -46,6 +75,10 @@ def run(**kw):
             **kw,
         )
 
+        mirrormode = mirror1.get_mirror_mode(imagespec)
+        # Verification of Negative test as image should not allow write operation from secondary
+        test_write_from_secondary(mirror2, mirrormode, imagespec)
```
Comment (manasagowri) [inline tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py:80]: The return value from this method should be returned back to run() method. So I guess you need to write something like

if test_write_from_secondary(mirror2, mirrormode, imagespec):
   return 1

File: tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py [hunk 2]
```diff
@@ -67,16 +102,56 @@ def run(**kw):
                 imagespec=imagespec,
                 state_pattern="up+replaying",
             )
-
-        # Cleans up the configuration
-        mirror1.clean_up(peercluster=mirror2, pools=[poolname])
+        # Verification of Negative test as image should not allow write operation from secondary
+        test_write_from_secondary(mirror2, mirrormode, imagespec)
```
Comment (manasagowri) [inline tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py:106]: The return value from this method should be returned back to run() method. So I guess you need to write something like

if test_write_from_secondary(mirror2, mirrormode, imagespec):
return 1

**Review by manasagowri — CHANGES_REQUESTED** [review]

File: tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py (hunk 1 above)
Comment (sunilangadi2) [reply tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py]: Done

File: tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py (hunk 2 above)
Comment (sunilangadi2) [reply tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py]: Done

**Comment by sunilangadi2** [comment]
@manasagowri Addressed review comments,
Test result for the new code change:
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-AFKZ49
Please review it.

**Review by manasagowri — APPROVED** [review]
LGTM group review done

**Pull request merged**